mcwb/volume.py

```diff
@@ -0,0 +1,131 @@
+from mcipc.rcon.enumerations import Item
+from mcipc.rcon.errors import InvalidArgument
+from mcipc.rcon.je import Client
+
+from mcwb import Anchor, Anchor3, Direction, Vec3, mktunnel
+
+__all__ = ['Volume']
+
+
+class Volume:
+    """
+    Describes a 3d space in a Minecraft world using a starting point and
+    size. The starting point can be any vertex or the middle of one of the
+    horizontal faces, using the cardinal terminology defined in Anchor3.
+    """
+
+    def __init__(
```

Comment:
The body of `__init__` is pretty huge and contains nested `if`s.
Please consider splitting it into different (class-) methods.

Reply:
in fda3abc I have gone for a minimal constructor and two factory methods one for each of the different approaches to defining a volume